src/components/Recipe/Form/Form.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { ReactElement, useState } from 'react';
+import useInputState from "./useInputState";
+import { useForm, SubmitHandler } from "react-hook-form";
+
+// @ts-ignore
+import { Link, useHistory } from 'react-router-dom'
+
+import { createRecipe, updateRecipe } from '../../../data/recipes/api';
+import { Button, ButtonSubmit } from '../../../styled'
+import { FormGroup, Label, LabelIngredient, Input, ErrorMessage } from './styled'
+
+import { Recipe } from '../../../data/recipes/types'
+
+interface Props {
+  initialRecipe?: Recipe;
+  initialId?: number;
+}
+
+interface RecipeFormValues {
+  name: string;
+  description: string;
+  ingredients: string[];
+}
+
+function Form({
+  initialRecipe = { name: '', description: '', ingredients: [{ name: '' }] },
+  initialId = 0
+}: Props): ReactElement {
+  const { register, handleSubmit, formState: { errors } } = useForm<RecipeFormValues>();
+  const[name, updateName] = useInputState(initialRecipe?.name || '');
+  const[description, updateDescription] = useInputState(initialRecipe?.description || '');
```

Comment:
The spacing is a bit inconsistent, consider using prettier (you'll use it a lot here at TK)